Add Imviz documentation (#842)
spacetelescope/jdaviz: docs/imviz/import_data.rst

.. _imviz-import-data:

***********
Import Data
***********

> Imviz can load data in the form of a filename (string), .jpg, .png,

Reviewer: ```suggestion
Imviz can load data in the form of a filename (FITS, JPEG, or PNG),
```

> an astropy object, or a numpy array if the data is 2D.

Reviewer: ```suggestion
an `~astropy.nddata.NDData` object, or a NumPy array if the data is 2D.
See :meth:`jdaviz.configs.imviz.helper.Imviz.load_data` for more information.
```

Importing data through the GUI
------------------------------

> The first way that users can load their data into the Imviz application is

Reviewer 2: In `displaying_images` you addressed the reader as "you", but here you're talking about "the user" in the abstract. I think you should use the former here for consistency.

by clicking the :guilabel:`Import` button at the top left of the application's
user interface. This opens a dialogue where the user can enter the path of file

> that can be parsed as a :class:`~astropy.nddata.NDData` (2D data only), :class:`~astropy.io.fits.HDUList`,

Reviewer: It only works on 2D, yes, but it is not limited to NDData, so it is misleading to put the parenthesis here. I think this was already covered in the section above anyway.

```suggestion
that can be parsed as a :class:`~astropy.nddata.NDData`, :class:`~astropy.io.fits.HDUList`,
```

> or :class:`~astropy.io.fits.ImageHDU` in the text field:
> 
> .. image:: ../cubeviz/img/cubeviz_import_data.png

Reviewer: I think @orifox said no need for screenshot. It is misleading to show Cubeviz here. Imviz user will see a different background and then they will be confused. If you must, refer back to relevant Cubeviz section with `:ref:` (as you have done below) but remove the image embedding.

```suggestion
or :class:`~astropy.io.fits.ImageHDU` in the text field.
```

After clicking :guilabel:`Import`, the data file will be parsed and loaded into the
application. A notification will appear to let users know if the data import
was successful. Afterward, the new data set can be found in the :guilabel:`Data`

> tab of each viewer's options menu as described in :ref:`cubeviz-selecting-data`

Reviewer: ```suggestion
tab of each viewer's options menu as described in :ref:`cubeviz-selecting-data`.
```

Importing data via the API
--------------------------

Alternatively, if users are working in a coding environment like a Jupyter
notebook, they have access to the Imviz helper class API. Using this API,

> users can load data into the application through code using the `load_data`

Reviewer: ```suggestion
users can load data into the application through code using the :meth:`~jdaviz.configs.imviz.helper.Imviz.load_data`
```

> method, which takes as input either the name of a local file or a

Reviewer: ```suggestion
method, which takes as input either the name of a local file, an
```

> :class:`~astropy.nddata.NDData` (2D data only), :class:`~astropy.io.fits.HDUList`,

Reviewer: ```suggestion
:class:`~astropy.nddata.NDData`, :class:`~astropy.io.fits.HDUList`,
```
